python_modules/dagster/dagster/components/core/tree.py

```diff
@@ -138,21 +143,34 @@ def from_module(
         )
 
     @classmethod
-    def load(cls, path_within_project: Path) -> Self:
+    def for_project(cls, project_root: Path) -> Self:
         """Using the provided path, find the nearest parent python project and load the
         ComponentTree using its configuration.
         """
-        from dagster_dg_core.context import DgContext
-
-        # replace with dagster_shared impl of path crawl and config resolution
-        dg_context = DgContext.for_project_environment(path_within_project, command_line_config={})
+        root_config_path = locate_dg_config_in_folder(project_root)
```

Comment:
ideally we should maintain the "find nearest parent python project"  flexibility as mentioned in docblock here to avoid breaking existing callsites that get invoked via clis

Reply:
think thats just moving to `discover_config_file`